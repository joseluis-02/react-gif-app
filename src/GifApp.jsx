import { useState } from 'react';
import { AgregarCategoria,GifGrid } from './componentes';

export const GifApp = () => {

    const [ categorias, setCategorias ] = useState(['One punch']);
    //console.log( categorias );
    const addCategoria = ( nuevoValor ) => {
        //console.log( nuevoValor );
        //setCategorias( c => [...c], 'Nuevo Valor');
        //prmera solucion
        //setCategorias([...categorias,'categorias']);
        //Segundo forma
        // Validar repetidos
        if ( categorias.includes(  nuevoValor ) ) return;

        setCategorias(c => [nuevoValor, ...categorias]);
    }

  return (
    <>
        {/* Titulo de mi aplicacion */}
        <h1>Buscador de Gifs</h1>

        {/*Input de la aplicai√≥n */}
        <AgregarCategoria 
            //Pasando una funcion a al componente hijo
            //setCategorias = { setCategorias }

            // el componente hijo retorna un valor nuevo
            onNewCategory = { valorRetorno => addCategoria(valorRetorno) }
        />

        {/*listado de Gifs o items */}
        { categorias.map( categoria => {
            return (
                <GifGrid
                    key={ categoria }
                    Categoria={ categoria }
                />
            )
            })
        }
        
    </>
  )
}
