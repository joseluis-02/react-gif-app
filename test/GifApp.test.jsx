import { render, screen } from "@testing-library/react";
import { GifApp } from "../src/GifApp";

describe('Prueba en mi componente principal GifApp', () => {
    test('Debe de hacer match con el snapshop', () => { 

        const { container } = render(
            <GifApp 

            />
        );

        //console.log( container );
        expect( container ).toMatchSnapshot();
        //screen.debug();
     });
});