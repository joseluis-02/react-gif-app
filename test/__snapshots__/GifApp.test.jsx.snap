// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en mi componente principal GifApp Debe de hacer match con el snapshop 1`] = `
<div>
  <h1>
    Buscador de Gifs
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    One punch
  </h3>
  <h4>
    Cargando...
  </h4>
  <div
    class="card-grid"
  />
</div>
`;
